yetanotherco/aligned_layer
# feat: various sdk improvements

# Sdk improvements

## Changes
- Forces the user to pass only one max fee for all the proofs instead of setting a max fee for each proof.
- Adds new message in batcher to retrieve the nonce from a given address.
- Adds get nonce from batcher in sdk.
- Adds a new aligned command to get the nonce.
- Remove the file for retrieving the nonce and queries it from the batcher.

Should be merged after #1325.

## Testing

To test it, setup a local devnet as usual and:
1. Send some proofs
2. Before they are sent to Ethereum run: 
```shell
cargo run --release -- get-user-nonce --user_addr "<YOUR_ADDRESS>"
```

If you are using the non_paying address use: `0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266` as the address.

You should see the same nonce as the batcher cache.

## Type of change

New feature

## Checklist
  - Closes #1309.
